src/azul/terra.py

```diff
@@ -195,10 +197,25 @@ class TDRClient(SAMClient):
     A client for the Broad Institute's Terra Data Repository aka "Jade".
     """
 
+    def project_for_source(self, source: TDRSource) -> str:
+        """
+        Use the TDR API to find which Google Cloud project contains the source.
+        """
+        response = self._get_source_response(source)
+        assert response.status == 200
+        response = json.loads(response.data)
+        return response['dataProject']
+
     def check_api_access(self, source: TDRSource) -> None:
         """
         Verify that the client is authorized to read from the TDR service API.
         """
+        response = self._get_source_response(source)
+        assert response.status == 200
+        log.info('TDR client is authorized for API access to %s: %r',
+                 source, response.data[:256].decode() + '...')
```

Comment:
I think we can rely on the response status code, without adding in any of the response. 
https://jade.datarepo-dev.broadinstitute.org/swagger-ui.html#/repository/retrieveSnapshot